app/javascript/Components/assignment_summary_table.jsx

```diff
@@ -30,16 +48,145 @@ export class AssignmentSummaryTable extends React.Component {
+      }),
+      columnHelper.accessor("group_name", {
+        id: "group_name",
+        header: () => I18n.t("activerecord.models.group.one"),
+        size: 100,
+        enableResizing: true,
+        cell: info => {
+          if (info.row.original.result_id) {
+            const path = Routes.edit_course_result_path(
+              this.props.course_id,
+              info.row.original.result_id
+            );
+            return (
+              <a href={path}>
+                {info.row.original.group_name}
+                {this.memberDisplay(info.row.original.group_name, info.row.original.members)}
+              </a>
+            );
+          } else {
+            return (
+              <span>
+                {info.row.original.group_name}
+                {this.memberDisplay(info.row.original.group_name, info.row.original.members)}
+              </span>
+            );
+          }
+        },
+      }),
+      markingStateColumn(this.state.marking_states, this.state.markingStateFilter),
+      columnHelper.accessor("tags", {
+        header: () => I18n.t("activerecord.models.tag.other"),
+        size: 90,
+        enableResizing: true,
+        cell: info => (
+          <ul className="tag-list">
+            {info.row.original.tags.map(tag => (
+              <li key={`${info.row.original._id}-${tag}`} className="tag-element">
+                {tag}
+              </li>
+            ))}
+          </ul>
+        ),
+        minWidth: 80,
+        enableSorting: false,
+        filterFn: (row, columnId, filterValue) => {
+          if (filterValue) {
+            // Check tag names
+            return row.original.tags.some(tag => tag.includes(filterValue));
+          } else {
+            return true;
+          }
+        },
+      }),
+      columnHelper.accessor("final_grade", {
+        header: () => I18n.t("results.total_mark"),
+        size: 100,
+        enableResizing: true,
+        cell: info => {
+          if (info.row.original.final_grade || info.row.original.final_grade === 0) {
+            const max_mark = Math.round(info.row.original.max_mark * 100) / 100;
+            return info.row.original.final_grade + " / " + max_mark;
+          } else {
+            return "";
+          }
+        },
+        meta: {className: "number"},
+        enableColumnFilter: false,
+        sortDescFirst: true,
+      }),
+    ];
+
+    const criteriaColumns = this.state.criteriaColumns.map(col =>
+      columnHelper.accessor(col.accessor, {
+        id: col.id,
+        header: () => col.Header,
+        size: 100,
+        enableResizing: true,
+        cell: value => value.getValue(),
+        meta: {className: "number"},
+        enableColumnFilter: false,
+      })
+    );
+
+    const bonusColumn = columnHelper.accessor("total_extra_marks", {
+      header: () => I18n.t("activerecord.models.extra_mark.other"),
+      size: 100,
+      enableResizing: true,
+      cell: info => info.getValue(),
```

Comment:
Even though you translated this correctly, I think this is a case where the original code was redundant, so you can just delete this `cell` attribute and use the react table default.